### PR TITLE
NoUnnecessaryTrailingUnderscore: Implement fix for let declarations

### LINKED ISSUE #9: NoUnnecessaryTrailingUnderscore: Implement fix for let declarations
https://github.com/jfmengels/elm-review-code-style/blob/d9c7f4556b5f6ddfd6d6298041c072f4e9109169/tests/NoUnnecessaryTrailingUnderscoreTest.elm#L463-L467

```elm
module A exposing (..)
a = let value_ = 1
    in value_
```

Should get a suggested fix:

```elm
module A exposing (..)
a = let value = 1
    in value
```

### PR DESCRIPTION
**Pull request by lydell**
Fixes https://github.com/jfmengels/elm-review-code-style/issues/9

… maybe. The tests pass. Are more tests needed?

### CODE REVIEW

File: src/NoUnnecessaryTrailingUnderscore.elm [hunk 1]
```diff
@@ -680,7 +670,7 @@ report patternsAndBody context =
                                 |> List.map (\var -> ( var.name, [] ))
                                 |> Dict.fromList
                     in
-                    { errors = List.filterMap (error expression (addNewScope names context.scopes)) declaredVariables
+                    { errors = List.filterMap (error expression Dict.empty (addNewScope names context.scopes)) declaredVariables
```
Comment (lydell) [inline src/NoUnnecessaryTrailingUnderscore.elm:668]: Is it correct to pass `Dict.empty` here?

Comment (jfmengels) [reply src/NoUnnecessaryTrailingUnderscore.elm:668]: I think it is. This didn't cause any issues on a large work project of mine at least

**Review by jfmengels — APPROVED** [review]
This seems to work well. Sorry for the delay, and thanks for looking at this 🙏

**Comment by jfmengels** [comment]
Released in v1.1.1

**Pull request merged**